src/internal/operators/finalize.ts

```diff
@@ -7,6 +7,43 @@ import { MonoTypeOperatorFunction, TeardownLogic } from '../types';
 /**
  * Returns an Observable that mirrors the source Observable, but will call a specified function when
  * the source terminates on complete or error.
+ * The specified function will also be called when the subscriber explicitly unsubscribes.
+ *
+ * ## Examples
+ * Execute callback function when the observable completes
+ *
+ * ```ts
+ * import { interval } from 'rxjs';
+ * import { take, finalize } from 'rxjs/operators';
+ *
+ * // emit value in sequence every 1 second
+ * const source = interval(1000);
+ * // 0,1,2,3,4,5....
+ * const example = source.pipe(
+ *   take(5), //take only the first 5 values
+ *   finalize(() => console.log('Sequence complete')) // Execute when the observable completes
+ * )
+ * const subscribe = example.subscribe(val => console.log(val));
+ * // 0,1,2,3,4,Sequence complete
+ * ```
+ *
+ * Execute callback function when the subscriber explicitly unsubscribes
+ *
+ * ```ts
+ * import { interval, timer, noop } from 'rxjs';
+ * import { finalize, tap } from 'rxjs/operators';
+ *
+ * const source = interval(100).pipe(
+ *   finalize(() => console.log('[finalize] Called')),
+ *   tap(noop, noop, () => console.log('[tap] Not called')),
+ * );
+ *
+ * const sub = source.subscribe(x => console.log(x), noop, () => console.log('[complete] Not called'));
```

Comment:
It's not particularly important in this example, but I would suggest passing `undefined` for the error callback. Passing a `noop` will swallow the error. Passing `undefined` will see any error thrown asynchronously as unhandled. I mention this because I think it would be good to avoid situations where a newcomer to RxJS might see this example and take up the pattern of passing `noop` and inadvertently introduce the swallowing of errors into their code.

Even better, though, pass an observer, like this:

```
...subscribe({
  next: value => /* whatever */,
  complete: () => /* whatever */
);
```